core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/report/detail/ReportDetailViewModel.kt

```diff
@@ -50,48 +63,85 @@ class ReportDetailViewModel(
                     text = systemImpl.getString(MR.strings.edit),
                     icon = FabUiState.FabIcon.EDIT,
                     onClick = this@ReportDetailViewModel::onClickEdit
-                )
+                ),
+                title = "Graph title",
             )
         }
-
-        launchIfHasPermission(
-            setLoadingState = true,
-            permissionCheck = { true }
-        ) {
-            launch {
-                navResultReturner.filteredResultFlowForKey(RESULT_KEY_REPORT_DETAIL)
-                    .collect { result ->
-                        val report = result.result as? Report ?: return@collect
-                        getReport(report)
+        viewModelScope.launch {
+            _uiState.whenSubscribed {
+                launchIfHasPermission(
+                    setLoadingState = true,
+                    permissionCheck = { true }
+                ) {
+                    val siteFlow =activeRepo.reportDao().findByUidLive(reportUid)
+                    launch {
+                        siteFlow.collect{
+                            val reportOptions = it?.let { optn ->
+                                optn.reportOptions?.takeIf { options -> options.isNotBlank() }
```

Comment:
As per [coding style](https://github.com/UstadMobile/UstadMobile/blob/primary/CODING-STYLE.md#general-coding-style) ```Do not use null checks that fail silently and would lead to code that doesn't behave as expect with no logging output or exception```

This is a null check that would silently fail and mislead. If something didn't load here; it's going to just makeup default ReportOptions2? That will just show wrong data for no apparent reason. Please update to handle in accordance with coding style.

Reply:
Note; this is not the only please this coding style point applies - please check all your code / viewmodels and ensure that 
a) You are not silently or incorrectly handling null values
b) You are not ignoring / catching exceptions that should be thrown; if an exception is caught and then nothing is shown to the user and/or developer that is a problem.